YarnSpinner/VirtualMachine.cs

```diff
@@ -117,60 +116,209 @@ internal enum TokenType {
         Text // a run of text until we hit other syntax
     }
 
-    internal class VirtualMachine
+    public abstract class SerializedVMState {
+
+        public string Text { get; protected set; }
+
+        public abstract VMState Deserialize();
+    }
+
+    public class JsonVMState : SerializedVMState
     {
+        public JsonVMState(string text) {
+            Text = text;
+        }
+
+        private VMState _deserialized;
+
+        public override VMState Deserialize() {
+            if (this._deserialized != null) {
+                return this._deserialized;
+            }
+
+            JsonParser parser = JsonParser.Default;
```

Comment:
Is there any reason you've used the JsonParser over the normal protobuf serialiser/deserialiser?

Reply:
On this I was thinking to provide SerializedVMState classes for JSON, XML, and proto, but I wasn't sure if it should be Yarn interest to provide multiple options so I decided to make JSON default as most people will be able to work with that.

If it would be preferred to default to proto that's fine, or should I make all 3 (2) options?

Reply:
Thank you :), I will address the problems as soon as I can.